form/form.go

```diff
@@ -0,0 +1,279 @@
+package form
+
+import (
+	"bytes"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const tagName = "form"
+
+type formOptions struct {
+	// IndexedArray indicates that contrary to standard "Rack-style" form
+	// encoding, array items should be index like `arr[0]=...&arr[1]=...`
+	// (normally it'd be `arr[]=...`).
+	IndexedArray bool
+
+	// Empty indicates that a field's value should be emptied in that its value
+	// should be an empty string. It's used to workaround the fact that an
+	// empty string is a string's zero value and wouldn't normally be encoded.
+	Empty bool
+
+	// Invert indicates that a boolean field's value should be inverted. False
+	// is the zero value for a boolean so it's convention in the library to
+	// specify a `No*` field to allow a false to be passed to the API. These
+	// fields should be annotated with `invert`.
+	Invert bool
```

Comment:
Ah! So it's sort of horrible, but we actually kind of need this given the current implementation. If you look at something like `AccountParams`:

``` go
type AccountParams struct {
...
	DebitNegativeBal     bool                          `form:"debit_negative_balances"`
	NoDebitNegativeBal   bool                          `form:"debit_negative_balances,invert"`
...
}
```

Without `NoDebitNegativeBal`, there's no way to detect whether a `false` was specified for `DebitNegativeBal` (because `false` is `bool`'s zero type), and therefore no way to set `DebitNegativeBal` to `false`. I've seen other SDKs user pointers (`*bool`) to work around this problem, which is awkward in its own way, but possibly better. I think for the time being we'll need to stick to this convention.